### ISSUE TITLE
Revise form saving progress UI

### ISSUE THREAD

**Issue by seadowg** [getodk/collect#5601]
https://github.com/getodk/collect/assets/967540/d239b1ef-ef34-4f7a-955f-17b07a96e51f

The dialog shown during saving (after the user hits "Save as Draft", "Finalize/Send" etc) will sometimes flash up on screen for a very short period of time and shows 4 messages for each stage of saving (Validating, writing to disk etc). This can make it feel like there was something important happening that the user might have missed. It'd be good to revise this flow so that it works better in cases where the form saves very quickly (< 500ms), but also supports longer form saves (could be as much as tens of seconds).

**Comment by lognaturel** [comment]
I think there might be two issues. The dialog text I believe is changed within the same dialog. That happens when saving at the end of a form too. 

The flashing from the screen capture is only using the exit dialog and it feels more like it could be pointing to a problem. We should at least check whether it happens in 2023.2.x

**Comment by lognaturel** [comment]
I've verified that the flashing happens in v2023.2.3 as well. It's possible I'm only noticing it now because I have a new-to-me Android Galaxy A13 which has a 90Hz refresh rate.

**Comment by lognaturel** [comment]
I still think there are two distinct issues here:
- The saving progress dialog shows many saving steps and the transition between them is very fast on a modern device. We could consider outputing fewer steps and/or only showing info about a step after it takes a certain amount of time.
- Something wacky is going on when using "Save as draft" from the form exit dialog. Maybe it's just that dialogs are stacking and then going away one at a time. Maybe what should happen is that the form exit dialog should go away after one of its buttons is tapped.

**Comment by seadowg** [comment]
> I still think there are two distinct issues here

Agreed. I'd like to bundle them up here though as I think it'll make sense to design and solve them together.
